guides/micronaut-java-records/micronaut-java-records.adoc

```diff
@@ -0,0 +1,193 @@
+include::{commondir}/common-header.adoc[]
+
+You are going to use https://docs.oracle.com/en/java/javase/16/language/records.html[Record Classes] in a Micronaut Application.
+
+____
+Record classes, which are a special kind of class, help to model plain data aggregates with less ceremony than normal classes.
+
+A record declaration specifies in a header a description of its contents; the appropriate accessors, constructor, equals, hashCode, and toString methods are created automatically. A record's fields are final because the class is intended to serve as a simple "data carrier".
+____
+
+include::{commondir}/common-requirements.adoc[]
+
+include::{commondir}/common-completesolution.adoc[]
+
+include::{commondir}/common-cli-or-launch.adoc[]
+
+[source,bash]
+----
+mn create-app --jdk=17 \
+   example.micronaut.micronautguide \
+   --features=data-jdbc,postgres,liquibase \ 
+   --build=@build@ \
+   --lang=@lang@ \
+----
+
+
+NOTE: To use Java records, specify JDK 17 when you create the application
+
+include::{commondir}/common-build-lang-arguments.adoc[]
+
+include::{commondir}/common-default-package.adoc[]
+
+NOTE: If you use https://launch.micronaut.io[Micronaut Launch], select "Micronaut Application" as application type and add the `postgres`, `data-jdbc` and `liquibase` as features.
+
+include::{commondir}/common-annotationprocessors.adoc[]
+
+=== Immutable Configuration with Java Records
+
+source:ValueAddedTaxConfiguration[]
+
+callout:configuration-properties[1]
+
+Write a test:
+
+test:ValueAddedTaxConfigurationTest[]
+
+callout:property[1]
+callout:micronaut-test-start-application-false[2]
+
+
+=== Database Migration with Liquibase
+
+We need a way to create the database schema. For that, we use
+https://micronaut-projects.github.io/micronaut-liquibase/latest/guide/[Micronaut integration with Liquibase].
+
+Add the following snippet to include the necessary dependencies:
+
+dependency:micronaut-liquibase[groupId=io.micronaut.liquibase]
+
+Configure the database migrations directory for http://www.liquibase.org[Liquibase] in `application.yml`.
+
+resource:application.yml[tag=liquibase]
+
+Create the following files with the database schema creation:
+
+resource:db/liquibase-changelog.xml[] 
+
+resource:db/changelog/01-create-books-schema.xml[] 
```

Comment:
This is not working. It's rendered as

![image](https://user-images.githubusercontent.com/559192/142606453-8573ae4f-e605-4eca-8299-71066adf702b.png)